core/templates/components/state-directives/outcome-editor/outcome-destination-editor.component.html

```diff
@@ -28,14 +28,17 @@
   </form>
   <p [hidden]="newStateNameForm.form.valid ||  outcomeNewStateName === undefined" class="form-text oppia-form-error oppia-form-error-text">Please pick a card name consisting of alphanumeric characters, spaces and/or hyphens.</p>
 
-  <div *ngIf="isSelfLoop() && canEditRefresherExplorationId" class="oppia-id-text">
+  <div *ngIf="isSelfLoop() && canEditRefresherExplorationId && outcome.refresherExplorationId != null" class="oppia-id-text">
```

Comment:
Just to check, if the user changes the refresher id to be empty, does it work as intended (e..g if `outcome.refresherExplorationId` is an empty string)?